test/TestGlobalFunctions.cpp

```diff
@@ -0,0 +1,46 @@
+#include "TestGlobalFunctions.h"
+
+
+Eigen::MatrixXd convertMFEMDenseToEigen(DenseMatrix* mat)
+{
+	auto res = Eigen::MatrixXd(mat->Width(), mat->Height());
+	for (int i = 0; i < mat->Width(); i++) {
+		for (int j = 0; j < mat->Height(); j++) {
+			res(i, j) = mat->Elem(i, j);
+		}
+	}
+	return res;
+}
+
+Eigen::MatrixXd buildMassMatrixEigen(std::unique_ptr<FiniteElementSpace>& fes)
+{
+	ConstantCoefficient one(1.0);
+	BilinearForm res(fes.get());
+	res.AddDomainIntegrator(new MassIntegrator(one));
+	res.Assemble();
+	res.Finalize();
+
+	return convertMFEMDenseToEigen(res.SpMat().ToDenseMatrix());
+}
+
+Eigen::MatrixXd buildInverseMassMatrixEigen(std::unique_ptr<FiniteElementSpace>& fes)
```

Comment:
if you are not transferring ownership, it is better to pass just a reference to fes or a raw pointer.

Reply:
Passing raw pointers instead.